### PR TITLE
Create netlify.yml

### PR DESCRIPTION
**Pull request by eukaryo**
I want to be able to run visual regression tests in PRs of this repository.
The function of netlify.yml is, only at the moment a PR is labeled as `safe-to-run-visual-regression-tests`, to run visual regression tests and deploy results to Netlify.

### CODE REVIEW

**Comment by eukaryo** [comment]
We will be able to see the results like the PR example below.
https://github.com/eukaryo/optuna-visualization-regression-tests/pull/1

**Comment by toshihikoyanase** [comment]
[Notes] 
- `netlify.yml` is created based on https://github.com/optuna/optuna/blob/master/.github/workflows/visual-regression.yml.
- `safe-to-run-visual-regression-tests` has not been created yet.

File: .github/workflows/netlify.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
+
+    - name: Install Optuna
+      run: |
+        pip install -U pip
+        pip install --progress-bar off .
+        git clone https://github.com/optuna/optuna.git
+        pip install optuna/
```
Comment (toshihikoyanase) [inline .github/workflows/netlify.yml:37]: A nit, but I think we can simplify these lines.
```suggestion
        pip install git+https://github.com/optuna/optuna.git
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. The proposed CI will accelerate the review process!

I have a small comment. Please take a look.

@c-bata What do you think of this PR?

File: .github/workflows/netlify.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
+
+    - name: Install Optuna
+      run: |
+        pip install -U pip
+        pip install --progress-bar off .
+        pip install git+https://github.com/optuna/optuna.git
+    - name: Install Optuna visual regression
```
Comment (c-bata) [inline .github/workflows/netlify.yml:37]: nit: I think you can use actions/cache like below to cache [the dataset of FasionMNIST](https://github.com/optuna/optuna-visualization-regression-tests/blob/5182483cb84c6c3dd401573cf98d8f197c83a174/studies.py#L244-L255).

https://github.com/optuna/optuna-visualization-regression-tests/blob/5182483cb84c6c3dd401573cf98d8f197c83a174/.github/workflows/gh-pages.yml#L24-L29

File: .github/workflows/netlify.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+         ( github.event_name == 'workflow_dispatch' )
+      )
+
+    steps:
+
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
```
Comment (c-bata) [inline .github/workflows/netlify.yml:30]: nit: How about using the newer Python version?

**Review by c-bata — APPROVED** [review]
Thank you for your pull request! Changes almost look good to me. I left some minor comments though.

File: .github/workflows/netlify.yml (hunk 2 above)
Comment (toshihikoyanase) [reply .github/workflows/netlify.yml:37]: Ah, this step actually cache the FashionMNIST dataset, not kurobako CLI. The step name may be a bit confusing...

**Comment by toshihikoyanase** [comment]
I agree with c-bata. @eukaryo What do you think of their comments?

File: .github/workflows/netlify.yml (hunk 3 above)
Comment (eukaryo) [reply .github/workflows/netlify.yml]: Agreed, I changed the version to 3.11.

**Comment by eukaryo** [comment]
I also agree. As for the use of the cache, however, I think we can prioritize the merge and do it later with another follow-up PR

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update.

> As for the use of the cache, however, I think we can prioritize the merge and do it later with another follow-up PR.

Sounds reasonable. LGTM!

**Pull request merged**